src/Compilers/VisualBasic/Portable/Symbols/Metadata/PE/PENamedTypeSymbol.vb

```diff
@@ -1556,6 +1557,94 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Symbols.Metadata.PE
         Friend NotOverridable Overrides Function GetSynthesizedWithEventsOverrides() As IEnumerable(Of PropertySymbol)
             Return SpecializedCollections.EmptyEnumerable(Of PropertySymbol)()
         End Function
+
+        Private ReadOnly Property HasAnyDeclaredRequiredMembers As Boolean
+            Get
+                Return ContainingPEModule.Module.HasAttribute(Handle, AttributeDescription.RequiredMemberAttribute)
```

Comment:
>RequiredMemberAttribute [](http://example.com/codeflow?start=91&length=23)

Can this attribute be applied in source?

Reply:
It can, but as with the other similar question, we aren't checking it. We're just not forbidding it.

Reply:
> It can, but as with the other similar question, we aren't checking it. We're just not forbidding it.

I think **not forbidding it** is going to complicate implementation for retargeting scenario. If one can apply attributes in source, they will be respected from metadata after we emit. Retargeting should "duplicate" this behavior, i.e. understand the attributes in source is if we emitted them to metadata and then imported them. I think it will be easier (and better) to forbid the attributes in source, unless we are willing to do the implementation and testing for retargeting.